app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -365,10 +385,61 @@ class SettingsFragment : PreferenceFragmentCompat(), CoroutineScope, AccountObse
 
     private fun updateAccountProfile(profile: Profile) {
         launch {
-            val preferenceFirefoxAccount =
-                findPreference<Preference>(context!!.getPreferenceKey(pref_key_account))
-            preferenceFirefoxAccount?.title = profile.displayName.orEmpty()
-            preferenceFirefoxAccount?.summary = profile.email.orEmpty()
+            context?.let { context ->
+                val preferenceFirefoxAccount =
+                    findPreference<AccountPreference>(context.getPreferenceKey(pref_key_account))
+
+                preferenceFirefoxAccount?.title?.setTextColor(
+                    R.attr.primaryText.getColorFromAttr(context)
+                )
+                preferenceFirefoxAccount?.title?.text = profile.displayName.orEmpty()
+                preferenceFirefoxAccount?.title?.visibility =
+                    if (preferenceFirefoxAccount?.title?.text.isNullOrEmpty()) View.GONE else View.VISIBLE
+
+                preferenceFirefoxAccount?.summary?.setTextColor(
+                    R.attr.primaryText.getColorFromAttr(context)
+                )
+                preferenceFirefoxAccount?.summary?.text = profile.email.orEmpty()
+
+                preferenceFirefoxAccount?.icon = ContextCompat.getDrawable(context, R.drawable.ic_shortcuts)
+                preferenceFirefoxAccount?.errorIcon?.visibility = View.GONE
+                preferenceFirefoxAccount?.background?.background = null
+            }
+        }
+    }
+
+    private fun displayAccountErrorIfNecessary() {
+        if (!requireComponents.backgroundServices.accountManager.accountNeedsReauth()) { return }
```

Comment:
`requireComponents` may throw here. AccountObserver methods will get called on some IO thread; it's possible for `displayAccountErrorIfNecessary` to be called at a point when there is no longer `context` available.
In general you can't assume very much about the state you're in when the registry decides to notify you of some event, it's much better to be defensive here.